azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -566,8 +566,13 @@ func resourceLinuxVirtualMachineRead(d *schema.ResourceData, meta interface{}) e
 			return fmt.Errorf("Error setting `secret`: %+v", err)
 		}
 	}
-
-	d.Set("priority", string(props.Priority))
+	// Resources created with azurerm_virtual_machine have priority set to ""
+	// We need to treat "" as equal to "Regular" to allow migration azurerm_virtual_machine -> azurerm_linux_virtual_machine
+	priority := props.Priority
+	if priority == "" {
+		priority = compute.Regular
+	}
+	d.Set("priority", string(priority))
```

Comment:
Could we please do this by assigning the default value first, and then check if response is empty and then assign the real value if it is empty.
```suggestion
	priority := string(compute.Regular)
	if props.Priority != "" {
		priority = string(props.Priority)
	}
	d.Set("priority", priority)
```